### PR TITLE
feat: specify custom UUID generation function

### PR DESCRIPTION
**Pull request by stewartmcgown**
### Description of change

In order to support UUID functions like sequential UUID, snowflake or even just uuid_generate_v1() we need a custom UUID generation function.

### Pull-Request Checklist

- [x] Code is up-to-date with the `master` branch
- [x] `npm run lint` passes with this change
- [x] `npm run test` passes with this change
- [ ] This pull request links relevant issues as `Fixes #0000`
- [ ] There are new or updated unit tests validating the change
- [x] Documentation has been updated to reflect this change
- [x] The new commits follow conventions explained in [CONTRIBUTING.md](https://github.com/typeorm/typeorm/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by stewartmcgown** [comment]
test failed with a segfault? hopefully not do with the pr itself!

File: docs/connection-options.md [hunk 1]
```diff
@@ -180,6 +180,8 @@ See [SSL options](https://github.com/mysqljs/mysql#ssl-options).
 
 * `uuidExtension` - The Postgres extension to use when generating UUIDs. Defaults to `uuid-ossp`. Can be changed to `pgcrypto` if the `uuid-ossp` extension is unavailable.
 
+* `uuidExtension` - The Postgres function to use when generating UUIDs. Overrides uuidExtension if provided.
```
Comment (AlexMesser) [inline docs/connection-options.md:183]: should be `uuidFunction`

**Comment by AlexMesser** [comment]
@stewartmcgown  can you please add a test showing that different custom functions works properly with this change?

**Comment by pleerock** [comment]
Closing as there is no answer from author. Feel free to re-open with required changes.

**Pull request closed without merging**